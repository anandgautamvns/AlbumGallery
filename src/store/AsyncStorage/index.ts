import AsyncStorage from '@react-native-async-storage/async-storage';
import {Album} from '../../services/type';

const ALBUMS_KEY = 'albums_list';

export const storeAlbums = async (list: Album[]) => {
  try {
    await AsyncStorage.setItem(ALBUMS_KEY, JSON.stringify(list));
  } catch {
    /* ignore */
  }
};

export const getStoredAlbums = async (): Promise<Album[] | null> => {
  try {
    const raw = await AsyncStorage.getItem(ALBUMS_KEY);
    return raw ? JSON.parse(raw) : null;
  } catch {
    return null;
  }
};

export const removeAlbums = async () => {
  try {
    await AsyncStorage.removeItem(ALBUMS_KEY);
  } catch {
    return null;
  }
};
