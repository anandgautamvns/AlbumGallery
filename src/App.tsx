import {QueryClient, QueryClientProvider} from '@tanstack/react-query';
import React, {Suspense, lazy} from 'react';
import {StatusBar, useColorScheme} from 'react-native';
import {SafeAreaProvider} from 'react-native-safe-area-context';
import {Colors} from 'react-native/Libraries/NewAppScreen';
import {Provider} from 'react-redux';
import Loading from './components/Loader';
import store from './store/redux/configureStore';
const Main = lazy(() => import('./main'));

const App: React.FC = () => {
  const queryClient = new QueryClient();
  const isDarkMode = useColorScheme() === 'dark';

  const backgroundStyle = {
    backgroundColor: isDarkMode ? Colors.darker : Colors.lighter,
  };

  return (
    <Provider store={store}>
      <QueryClientProvider client={queryClient}>
        <Suspense fallback={<Loading />}>
          <StatusBar
            barStyle={isDarkMode ? 'light-content' : 'dark-content'}
            backgroundColor={backgroundStyle.backgroundColor}
          />
          <SafeAreaProvider>
            <Main backgroundStyle={backgroundStyle} />
          </SafeAreaProvider>
        </Suspense>
      </QueryClientProvider>
    </Provider>
  );
};

export default App;
