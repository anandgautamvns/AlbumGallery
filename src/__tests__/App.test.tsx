import {QueryClient, QueryClientProvider} from '@tanstack/react-query';
import {render} from '@testing-library/react-native';
import React from 'react';
import {Provider} from 'react-redux';
import store from '../store/redux/configureStore';
import App from './../App';

jest.mock('../main', () => {
  return () => <></>; // Mock the Main component
});

jest.mock('react-native/Libraries/NewAppScreen', () => ({
  Colors: {
    darker: '#000000',
    lighter: '#ffffff',
  },
}));

describe('App Component', () => {
  it('renders correctly', () => {
    const queryClient = new QueryClient();
    const {toJSON} = render(
      <Provider store={store}>
        <QueryClientProvider client={queryClient}>
          <App />
        </QueryClientProvider>
      </Provider>,
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
