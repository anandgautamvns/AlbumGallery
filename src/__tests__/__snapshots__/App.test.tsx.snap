// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
/>
`;
